src/Bicep.LangServer/Handlers/BicepGenerateParamsCommandHandler.cs

```diff
@@ -0,0 +1,147 @@
+using Bicep.Core.Semantics;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.Workspaces;
+using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Utils;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    // This handler is used to generate compiled parameters.json file for given a bicep file path.
+    // It returns generate-params succeeded/failed message, which can be displayed approriately in IDE output window
+    public class BicepGenerateParamsCommandHandler : ExecuteTypedResponseCommandHandlerBase<string, string>
+    {
+        private readonly ICompilationManager compilationManager;
+        private readonly EmitterSettings emitterSettings;
+        private readonly IFeatureProvider features;
+        private readonly IFileResolver fileResolver;
+        private readonly IModuleDispatcher moduleDispatcher;
+        private readonly INamespaceProvider namespaceProvider;
+        private readonly IConfigurationManager configurationManager;
+
+        public BicepGenerateParamsCommandHandler(ICompilationManager compilationManager, ISerializer serializer, IFeatureProvider features, EmitterSettings emitterSettings, INamespaceProvider namespaceProvider, IFileResolver fileResolver, IModuleDispatcher moduleDispatcher, IConfigurationManager configurationManager)
+            : base(LangServerConstants.GenerateParamsCommand, serializer)
+        {
+            this.compilationManager = compilationManager;
+            this.emitterSettings = emitterSettings;
+            this.features = features;
+            this.namespaceProvider = namespaceProvider;
+            this.fileResolver = fileResolver;
+            this.moduleDispatcher = moduleDispatcher;
+            this.configurationManager = configurationManager;
+        }
+
+        public override Task<string> Handle(string bicepFilePath, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(bicepFilePath))
+            {
+                throw new ArgumentException("Invalid input file path");
+            }
+
+            DocumentUri documentUri = DocumentUri.FromFileSystemPath(bicepFilePath);
+            string output = GenerateCompiledParametersFileAndReturnOutputMessage(bicepFilePath, documentUri);
+
+            return Task.FromResult(output);
+        }
+
+        private string GenerateCompiledParametersFileAndReturnOutputMessage(string bicepFilePath, DocumentUri documentUri)
+        {
+            string compiledFilePath = PathHelper.ResolveParametersFileOutputPath(bicepFilePath);
+            string compiledFile = Path.GetFileName(compiledFilePath);
+
+            // If the template exists and contains bicep generator metadata, we can go ahead and replace the file.
```

Comment:
TODO

Reply:
Also, it shows a validation error in the vscode editor:
![image](https://user-images.githubusercontent.com/6913354/171500780-6eed8a11-d9cc-4380-8b1a-339be78b57ae.png)

Reply:
to match the generated json and parameters file, I add the metadata into the parameters file too. per my tests, azure cli successfully created deployment, even if the parameters file has metadata field in it.

I couldn't understand what you mean by `if you start with your own params file you can't use this command`, it must parse the existing parameters file, and use the `values`, `contentVersion`, etc. off of it.

you're right about the vscode extension, though.

@ucheNkadiCode , what you think?

Reply:
@ucheNkadiCode Let's discuss in person at our sync.

Reply:
Remove metadata.

Reply:
done! metadata field is removed. tests are updated to reflect changes. $schema field used to validate the generated json file, instead of metadata field.